client/boot/index.js

```diff
@@ -323,7 +323,11 @@ function reduxStoreReady( reduxStore ) {
 			} );
 
 			if ( '/' === context.pathname && config.isEnabled( 'devdocs/redirect-loggedout-homepage' ) ) {
-				page.redirect( '/devdocs/start' );
+				if ( config.isEnabled( 'oauth' ) ) {
+					page.redirect( '/authorize' );
+				} else {
+					page.redirect( '/devdocs/start' );
```

Comment:
@mtias, currently we only redirect if `devdocs` is enabled. Or at least implicitly, since the name of the feature flag includes `devdocs` and in the current config `devdocs/redirect-loggedout-homepage` is only enabled if `devdocs` is enabled (not true in the other direction, see `wpcalypso`).

If we want to cover the more generic case, does this flow make sense?

- redirect to generic `/authorize` if **OAuth is enabled** and **DevDocs is disabled**
- redirect to a custom `/authorize` screen, which redirects to `/devdocs/welcome` when successful and links to `/start/developer` to register if **both OAuth and DevDocs are enabled**
- redirect to `/devdocs/start` if **OAuth is disabled** and **DevDocs is enabled**
- redirect to generic WordPress.com login/register screen if **both OAuth and DevDocs are disabled** (should never happen in production, because of the way Calypso is proxied, but still a good fallback case)